### PR TITLE
chat update and delete method with postman tests

### PR DESCRIPTION
**Pull request by Devesh21700Kumar**
setup the Update and delete by ID chat methods in the controllers

Below are the Postman tests:-

1). Update by ID

image

2). Delete by ID

image

image

### CODE REVIEW

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09  pls see this one

File: src/chat/chat.controller.ts [hunk 1]
```diff
@@ -89,8 +88,8 @@ export class ChatController {
   }
 
   // Delete a Chat
-  @Delete('/delete')
-  async deleteChat(@Res() res, @Query('uid') uid) {
+  @Delete('/:uid')
+  async deleteChat(@Res() res, @Param('uid') uid: string) {
```
Comment (Abhishek-kumar09) [inline src/chat/chat.controller.ts:92]: Should delete have only uid?
Like when we delete chat we delete some message and one uid can have many message.

This will delete whole of the chat, again that can be the use case, so please create a new controller for deleteing some specific chat by id

File: src/user/user.service.ts [hunk 2]
```diff
@@ -38,7 +38,10 @@ export class UserService {
   }
 
   // Edit User details
-  async updateUser(UserID, CreateUserDTO: CreateUserDTO): Promise<User> {
+  async updateUser(
+    UserID,
+    CreateUserDTO: CreateUserDTO,
+  ): Promise<User | undefined> {
```
Comment (Abhishek-kumar09) [inline src/user/user.service.ts:44]: Why this change?

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]

File: src/chat/chat.controller.ts (hunk 1 above)
Comment (Devesh21700Kumar) [reply src/chat/chat.controller.ts:92]: Yeah, . will do that  , thanks

File: src/user/user.service.ts (hunk 2 above)
Comment (Devesh21700Kumar) [reply src/user/user.service.ts:44]: The promise might return either user type or undefined type, that's why I thought of doing it.

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09 
1).chatId init has been done
2).@Res decorator res method has been removed

**Comment by Abhishek-kumar09** [comment]
Resolve merge conflicts

**Comment by Devesh21700Kumar** [comment]
> Resolve merge conflicts

@Abhishek-kumar09  resolved conflicts...

File: src/modules/chat/chat.controller.ts [hunk 3]
```diff
@@ -36,32 +35,45 @@ export class ChatController {
   // add a Chat
   @Post()
   @UsePipes(ValidationPipe)
-  async addChat(@Res() res, @Body() CreateChatDTO: CreateChatDTO) {
-    return await this.chatService.addChat(CreateChatDTO);
+  async addChat(@Body() CreateChatDTO: CreateChatDTO) {
```
Comment (Abhishek-kumar09) [inline src/modules/chat/chat.controller.ts:38]: revert this

File: src/modules/chat/chat.controller.ts [hunk 4]
```diff
@@ -36,32 +35,45 @@ export class ChatController {
...
+    const chat = await this.chatService.addChat(CreateChatDTO);
+    return chat;
   }
 
   // Retrieve Chats list
   @ApiCreatedResponse({ type: [ChatResponseBody] })
   @Get()
   async getAllChat() {
-    return await this.chatService.getAllChat();
+    const chats = await this.chatService.getAllChat();
```
Comment (Abhishek-kumar09) [inline src/modules/chat/chat.controller.ts:47]: revert this

File: src/modules/chat/chat.controller.ts [hunk 5]
```diff
@@ -36,32 +35,45 @@ export class ChatController {
...
+    return chats;
   }
 
   // Fetch a particular Chat using ID
   @ApiCreatedResponse({ type: ChatResponseBody })
   @Get('/:chatId')
   async getChat(@Param('chatId') chatId: string) {
-    return await this.chatService.getChat(chatId);
+    const chat = await this.chatService.getChat(chatId);
```
Comment (Abhishek-kumar09) [inline src/modules/chat/chat.controller.ts:55]: revert this

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]
Squash all your commits into one and then rebase your branch. New changes are not reflected in your code.

File: src/modules/chat/chat.controller.ts [hunk 6]
```diff
@@ -19,25 +16,16 @@ import { ApiCreatedResponse, ApiProperty } from '@nestjs/swagger';
...
-
-  @ApiProperty({ required: true, example: '+919999999999' })
-  phone: string;
-
-  @ApiProperty({ required: true, example: 'A-88, Mayur Vihar, Delhi' })
-  address: string;
-
-  @ApiProperty({ required: true, example: 'I am Noob Coder' })
-  description: string;
+  chats: string;
```
[the remark this answers is not in the document]
Comment (Devesh21700Kumar) [reply src/modules/chat/chat.controller.ts]: i have also changed create chat dto , entity and model accordingly

**Comment by Devesh21700Kumar** [comment]
changed
